### ISSUE TITLE
Recent song history

### ISSUE THREAD

**Issue by CaspianA1** [WBOR-91-1-FM/wbor-studio-dashboard#27]

**Comment by mdrxy** [comment]
<img width="505" alt="Screenshot" src="https://github.com/user-attachments/assets/ba831ab8-df9a-4215-aeea-4e0bd3a7d95b">

Could be similar to this

**Comment by CaspianA1** [comment]
Regarding this, basically 80% of the Twilio code is built around maintaining a list of items (associated with textures) sorted by history, while allowing for arbitrary insertions and deletions (e.g. a message in the middle of the history being deleted) without any performance degradation - so no re-requesting for the same message (or texture rebuilding) at any point

So that code could be made generic and used for song history stuff too, and then reused for this - it would just take some time

**Comment by mdrxy** [comment]
Something to consider regarding this is updates to spins that are made after initially posting them. I'm thinking about scenarios where the auto-logging may incorrectly log the song, and then a DJ comes in after to correct it. At each song update, perhaps it should refresh the song history log to catch these cases?

**Comment by CaspianA1** [comment]
Yeah the Twilio code handles this too - e.g. for Twilio, deleted/changed messages are properly amended
So it could be adapted for this purpose w/ Spinitron

Just haven't done it yet - would take a couple of days to implement well

**Comment by mdrxy** [comment]
Same as in #14, Spinitron will POST to our relay server each spin, and spins will be sent as messages in the MQ exchange consumed by the dashboard to build up a local history of recent spins that this could pull from.

**Comment by CaspianA1** [comment]
Implemented on [this](https://github.com/WBOR-91-1-FM/wbor-studio-dashboard/tree/CaspianA1/api_history_list) branch (not merged to `main` yet).
